R/grade_this.R

```diff
@@ -269,6 +274,24 @@ NULL
 #' @export
 .user_code <- placeholder(".user_code")
 
+empty_code <- function(code, check_null = FALSE) {
```

Comment:
I think this function is trying to do too much. Instead of testing for empty code _and_ creating the empty code grade, I think we should do it in two steps.

```r
check_is_code_empty <- function(code, check_null) {
  # returns TRUE or FALSE
}

grade_code_is_empty <- function(...) {
  graded(logical(), "message", type = "info")
}
```

so you would use it like

```r
if (check_is_code_empty(user_code)) {
  return(grade_code_is_empty())
}
```

Reply:
Those two functions should be higher up, perhaps under `detect_grade_this()`, since this section covers the placeholder objects.

Reply:
Split and relocated in [033315d](https://github.com/rstudio/gradethis/pull/288/commits/033315de49b6da55ef73c1f79916c30d7ed79774)